bentoml/_internal/utils/analytics/usage_stats.py

```diff
@@ -0,0 +1,170 @@
+    # cached for better performance.
+    return int(os.environ.get(BENTOML_USAGE_REPORT_INTERVAL_SECONDS, 12 * 60 * 60))
+
+
+def get_serve_info() -> t.Dict[str, str]:  # pragma: no cover
+    # Returns a safe token for serve as well as timestamp of creating this token
+    return {
+        "serve_id": secrets.token_urlsafe(32),
+        "serve_creation_timestamp": datetime.now(timezone.utc).isoformat(),
+    }
+
+
+@lru_cache(maxsize=1)
+def get_platform() -> str:
+    return platform.platform(aliased=True)
+
+
+@lru_cache(maxsize=1)
+def get_python_version() -> str:
+    return platform.python_version()
+
+
+@lru_cache(maxsize=1)
+def get_client_id() -> t.Dict[str, str]:  # pragma: no cover
+    if os.path.exists(CLIENT_ID_PATH):
+        with open(CLIENT_ID_PATH, "r", encoding="utf-8") as f:
+            client_id = yaml.safe_load(f)
+        return client_id
+    return {}
+
+
+def get_hardware_usage(pid: int) -> t.Dict[str, t.Any]:  # pragma: no cover
+    proc = psutil.Process(pid)
+    with proc.oneshot():
+        return {
+            "memory_usage_percent": proc.memory_percent(),
+            "total_memory(MB)": psutil.virtual_memory().total / 1024**2,
+            "num_threads": proc.num_threads(),
+        }
+
+
+@inject
+def get_payload(
+    event_type: str,
+    event_pid: int,
+    event_properties: t.Optional[t.Dict[str, t.Any]] = None,
+    session_id: str = Provide[BentoMLContainer.session_id],
+) -> t.Dict[str, t.Any]:
+    """
+    event_type(str): type of event (model_save, bento_build, etc.)
+    event_pid(int): retrieved with os.getpid() for the given event
+    event_properties(dict(str, Any)): custom properties of a given event
+    """
+    if event_properties is None:
+        event_properties = {}
+
+    # TODO: add yatai_user email and yatai_version when available
+    common_stats = {
+        "bentoml_version": BENTOML_VERSION,
+        "python_version": get_python_version(),
+        "platform": get_platform(),
+        "session_id": session_id,
+        **get_client_id(),
+    }
+
+    return {
+        "event_type": event_type,
+        **event_properties,
+        **common_stats,
+        **get_hardware_usage(event_pid),
+    }
+
+
+@slient
+def send_usage_event(payload: t.Dict[str, t.Any], timeout: int, uri: str) -> None:
+    requests.post(uri, json=payload, timeout=timeout)
+
+
+@slient
+def track(
+    event_type: str,
+    event_pid: int,
+    event_properties: t.Optional[t.Dict[str, t.Any]] = None,
+    timeout: int = 2,
+    uri: str = BENTOML_TRACKING_URL,
+) -> None:
+    if do_not_track():
+        return
+    payload = get_payload(
+        event_type=event_type, event_pid=event_pid, event_properties=event_properties
+    )
+    send_usage_event(payload, timeout=timeout, uri=uri)
+
+
+@slient
+def scheduled_track(
+    event_pid: int,
+    event_properties: t.Optional[t.Dict[str, t.Any]] = None,
+    interval: int = get_usage_stats_interval_seconds(),
```

Comment:
let's just fix this to BENTOML_USAGE_REPORT_INTERVAL_SECONDS, I don't think we should allow user to customize this. Fixing it also simplifies our analytic query.

Reply:
see you want to set it in testing, made a suggestion above to patch the constant variable in testing instead